spec/spec.md

```diff
@@ -0,0 +1,109 @@
+# Dapr SDK Spec - Version 1.9-alpha
+
+## Abstract
+
+Dapr SDK Spec is a language-neutral specification serving as a reference to Dapr SDKs.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Notations and Terminology](#notations-and-terminology)
+- [Versioning](#versioning)
+- [Client Object](#client-object)
+- [Object Serialization](#object-serialization)
+- [Exception Handling](#exception-handling)
+- [Core Building Blocks](#core-building-blocks)
+- [Actors](#Actors)
+- [Preview APIs](#preview-apis)
+
+## Overview
+
+[Dapr](https://github.com/dapr/dapr) offers a sidecar with an HTTP and gRPC API listening on localhost. In order to offer an additional layer of convenience, Dapr also offers SDKs in a variety of languages. Additionally, actors in Dapr rely on the SDK for applications to implement their actor classes and easy inocation of actor methods, timers and reminders.
+
+Each SDK has its own set of maintainers and decisions are made per repository. Over time, decisions per SDK maintainer can deviate from a consistent experience when using Dapr in each programming language. This specification's goal is to bring standards to the Dapr experience via SDKs while still allowing each SDK to be idiomatic to its programming language and target audience.
+
+## Notations and Terminology
+
+### Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Terminology
+
+This specification defines the following terms:
+
+#### SDK
+
+An "SDK" or "Software Development Kit" is defined as one or more libraries that offers a set of public interfaces, methods, classes and attributes offering a programming model for Dapr native to a given programming language.
+
+#### Domain Object
+
+A "domain object" is defined as an instance of a class from the application's code and not part of the SDK.
+
+## Versioning
+
+### Spec Versioning
+
+This spec follows [semantic versioning 2.0](https://semver.org/spec/v2.0.0.html). Matching the Dapr runtime API version for major and minor. It means that for each minor version release of Dapr runtime, there will be a new release of the spec. The patch version of the spec and runtime increase independently.
```

Comment:
The subtlety of this is it reads (to me) as you need to enforce keeping version numbers the same between SDK and runtime.  But I believe, and correct me, this is also trying to state there needs to be 1 to 1 parity at all times between SDK and runtime.  Could we please clarify?

Reply:
I reworded this to make it clear that the parity is between the spec and the runtime. The SDKs can still be versioned independently - so the table in docs that matches the SDK versions with runtime versions remains and the spec compliance is automatically done via the runtime version.